database/couchbase_bucket.py

```diff
@@ -0,0 +1,341 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: couchbase_bucket
+short_description: Idempotently creates or deletes Couchbase buckets.
+description:
+    - Creates or removes buckets on the Couchbase server using the Couchbase
+            REST API.
+        'state' = 'present' and 'force' = 'no', the bucket will be created
+            if it does not exist.
+        'state' = 'present' and 'force' = 'yes', the bucket will be deleted
+            and recreated if it does exist.
+        'state' = 'absent' and 'force' = 'no', the bucket will be deleted.
+        'state' = 'absent' and 'force' = 'yes', the bucket will be deleted.
+version _added: "1.7"
```

Comment:
This is clearly no longer right, but we can always change it if this gets closer